Sources/Logging/Logging.swift

```diff
@@ -653,55 +661,165 @@ extension Logger {
 
     /// `bootstrap` is a one-time configuration function which globally selects the desired logging backend
     /// implementation. `bootstrap` can be called at maximum once in any given program, calling it more than once will
     /// lead to undefined behavior, most likely a crash.
     ///
     /// - parameters:
+    ///     - metadataProvider: The `MetadataProvider` used to inject runtime-generated metadata, defaults to nil.
     ///     - factory: A closure that given a `Logger` identifier, produces an instance of the `LogHandler`.
-    public static func bootstrap(_ factory: @escaping (String) -> LogHandler) {
+    public static func bootstrap(_ factory: @escaping (String, Logger.MetadataProvider) -> LogHandler) {
         self._factory.replaceFactory(factory, validate: true)
     }
 
-    // for our testing we want to allow multiple bootstraping
+    /// `bootstrap` is a one-time configuration function which globally selects the desired logging backend
+    /// implementation. `bootstrap` can be called at maximum once in any given program, calling it more than once will
+    /// lead to undefined behavior, most likely a crash.
+    ///
+    /// - parameters:
+    ///     - factory: A closure that given a `Logger` identifier, produces an instance of the `LogHandler`.
+    public static func bootstrap(_ factory: @escaping (String) -> LogHandler) {
+        self._factory.replaceFactory({ label, _ in
+            factory(label)
+        }, validate: true)
+    }
+
+    /// `bootstrapMetadataProvider` is a one-time configuration function which globally selects the desired
+    /// ``Logger/MetadataProvider`` which should be used by all loggers created in this process from here onwards.
+    ///
+    /// `bootstrapMetadataProvider` can be called at maximum once in any given program, calling it more than once will
+    /// lead to undefined behavior, most likely a crash.
+    ///
+    /// It is also possible to `bootstrap(metadataProvider:_:)` to configure both a metadata provider,
+    /// and log handler backend at the same time, in a single call.
+    ///
+    /// - Parameter metadataProvider: The metadata provider to be used by this logging system.
+    public static func bootstrapMetadataProvider(_ metadataProvider: Logger.MetadataProvider) {
+        self._metadataProviderFactory.replaceMetadataProvider(metadataProvider, validate: true)
+    }
+
+    // for our testing we want to allow multiple bootstrapping
     internal static func bootstrapInternal(_ factory: @escaping (String) -> LogHandler) {
+        self._metadataProviderFactory.replaceMetadataProvider(.noop, validate: false) // TODO: fixme remove this and fix tests
+        self._factory.replaceFactory({ label, _ in
+            factory(label)
+        }, validate: false)
+    }
+
+    // for our testing we want to allow multiple bootstraping
+    internal static func bootstrapInternal(_ factory: @escaping (String, Logger.MetadataProvider) -> LogHandler) {
```

Comment:
bootstrapInternalMetadataProvider? or maybe collapse the two internal bootstrap functions to one, with metadata provider optional?